### PR TITLE
Sb 945 model service

### PR DESCRIPTION
**Pull request by manjudr**

### CODE REVIEW

File: src/app/helpers/announcement/ObjectStore/ObjectStoreRest.js [hunk 1]
```diff
@@ -80,63 +87,41 @@ class ObjectStoreRest extends ObjectStore {
...
+                    body: {
+                        request: {
+                            'tableName': data.table,
+                            'documentName': data.table,
+                            'payload': data.values
+                        }
+                    },
+                    json: true
+                }
                 let result = await (this.httpService(options))
```
Comment (kochhar) [inline src/app/helpers/announcement/ObjectStore/ObjectStoreRest.js]: why does this use its own httpservice? why not use the http wrapper which is used in the notification service also?

File: src/app/helpers/announcement/controller.js [hunk 2]
```diff
@@ -17,14 +16,11 @@ let ApiInterceptor = require('sb_api_interceptor')
...
 
-  constructor() {
-    //table name should be same as the name in database table
-    let tableMapping = {
-      'announcement': AnnouncementModel,
-      'announcementtype': AnnouncementTypeModel,
-      'attachment': AttachmentModel,
-      'metrics': MetricsModel
-    }
+  constructor({metrics, announcement, announcementtype } = {}) {
```
Comment (kochhar) [inline src/app/helpers/announcement/controller.js]: this constructor needs documentation to explain the object passed in and the properties which they must have.

File: src/app/helpers/announcement/model/AnnouncementCreateModel.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ */
+
+let BaseModel = require('./BaseModel.js')
+let Joi = require('joi')
+
+/**
+ * Announcement create model validation
+ * @extends BaseModel
+ */
+class AnnouncementModel extends BaseModel {
```
Comment (kochhar) [inline src/app/helpers/announcement/model/AnnouncementCreateModel.js]: this should have two additional methods: fromApi and fromDb

File: src/app/helpers/announcement/ObjectStore/ObjectStoreRest.js (hunk 1 above)
Comment (manjudr) [reply src/app/helpers/announcement/ObjectStore/ObjectStoreRest.js]: This is not in the part of notification service we have done this clean up in announcement model service task

Comment (kochhar) [reply src/app/helpers/announcement/ObjectStore/ObjectStoreRest.js]: great.

**Pull request merged**